dsgrid/config/project_config.py

```diff
@@ -386,6 +376,26 @@ def get_base_dimension(self, dimension_type: DimensionType):
                 return dim_config
         assert False, dimension_type
 
+    def get_dimension_records(self, dimension_type: DimensionType, query_name: str):
```

Comment:
@lixiliu I'm open to different names. I didn't include `dataframe` for brevity - and because I assume that anyone calling this will want a dataframe. That being said, we could return the list of data models instead.

Reply:
dataframe is better